lib/util/pools.js

```diff
@@ -78,10 +78,8 @@ export function createPool(name, opts) {
       }
     },
 
-    freeAll: function() {
-      let allocatedLength = allocated.length;
-      let freeLength = free.length;
-      let reAlloc = allocated.slice(0, size - freeLength);
+    freeAll: function freeAllPool() {
+      let reAlloc = allocated.slice(0, size - free.length);
```

Comment:
`size` is the current total size of any pool, by default it is 10k, but there has been some thinking about starting smaller and gradually increasing the size. This would help curb the initial cost of including diffHTML on the page.

`free` is the current amount of free objects available from the pool. I think you're right that this needs a `-1`. Otherwise we're probably leaving one element lingering, although none of the tests have caught this case.